File: SchoolEnglish.Domain/TaskPart.cs
namespace SchoolEnglish.Domain
{
    public class TaskPart
    {
        public Guid Id { get; set; }
        public string Number { get; set; }
        public string RightAnswer { get; set; }
        public string ClueText { get; set; }
        public int ClueCost { get; set; }
        public Guid TaskId { get; set; }
        public Task Task { get; set; }
        public TaskPartContent Content { get; set; }
        public IEnumerable<UserAnswer> Answers { get; set; }
    }
}

File: SchoolEnglish.UI/school_english/lib/models/task_part/taskpart.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:school_english/models/taskpart_content/taskpart_content.dart';

part 'taskpart.g.dart';

@JsonSerializable()
class TaskPart {
  final String id;
  final String number;
  final String rightAnswer;
  final String clueText;
  final int clueCost;
  final TaskPartContent content;
  final String taskId;

  const TaskPart(
      {required this.id,
      required this.number,
      required this.rightAnswer,
      required this.clueText,
      required this.clueCost,
      required this.content,
      required this.taskId});

  factory TaskPart.fromJson(Map<String, dynamic> json) =>
      _$TaskPartFromJson(json);

  Map<String, dynamic> toJson() => _$TaskPartToJson(this);
}
